flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartitionTest.java

```diff
@@ -161,6 +173,72 @@ public Void call() throws Exception {
 		blockingFinish.get();
 	}
 
+	/**
+	 * Tests a fix for FLINK-12544.
+	 *
+	 * @see <a href="https://issues.apache.org/jira/browse/FLINK-12544">FLINK-12544</a>
+	 */
+	@Test
+	public void testConcurrentRequestAndReleaseMemory() throws Exception {
```

Comment:
Is this test failing (at least from time to time) before your change?

Reply:
Yes, it could be failing every time after reverting my changes, because I control the logic sequence strictly.